Submitted By:            Douglas R. Reno <renodr at linuxfromscratch dot org>
Date:                    2022-07-11
Initial Package Version: 5.0.1
Origin:                  Arch Linux
Upstream Status:         Unknown
Description:             Adds the av_stream_get_first_dts() function so that
                         QtWebEngine (and other Chromium forks) can build with
                         system ffmpeg when the version is above ffmpeg-4.4.

diff -Naurp ffmpeg-5.0.1.orig/libavformat/avformat.h ffmpeg-5.0.1/libavformat/avformat.h
--- ffmpeg-5.0.1.orig/libavformat/avformat.h	2022-01-14 12:45:40.000000000 -0600
+++ ffmpeg-5.0.1/libavformat/avformat.h	2022-07-11 20:27:08.970971058 -0500
@@ -1115,6 +1115,10 @@ struct AVCodecParserContext *av_stream_g
  */
 int64_t    av_stream_get_end_pts(const AVStream *st);
 
+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff -Naurp ffmpeg-5.0.1.orig/libavformat/utils.c ffmpeg-5.0.1/libavformat/utils.c
--- ffmpeg-5.0.1.orig/libavformat/utils.c	2022-04-04 09:40:22.000000000 -0500
+++ ffmpeg-5.0.1/libavformat/utils.c	2022-07-11 20:28:00.811577200 -0500
@@ -92,6 +92,13 @@ int64_t av_stream_get_end_pts(const AVSt
         return AV_NOPTS_VALUE;
 }
 
+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+   return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 struct AVCodecParserContext *av_stream_get_parser(const AVStream *st)
 {
     return cffstream(st)->parser;
